src/graphql/resolveProcedure.js

```diff
@@ -34,13 +39,26 @@ const resolveProcedure = (procedure, getProcedureArgs) => {
   return async (source, args, { client }) => {
     const procedureArgs = getProcedureArgs(source, args)
 
-    // Actuall run the procedure using our arguments.
-    const result = await client.queryAsync({
-      name: query.name,
-      text: query.text,
-      values: argEntries.map(([name]) => procedureArgs[camelCase(name)]),
+    const values = argEntries.map(([name, type]) => {
+      const obj = procedureArgs[camelCase(name)]
+      let value
+
+      if (type.isTableType) {
+        value = reduce(obj, (acc, val, key) => {
```

Comment:
This was actually one of the first things I tried. However this does not work and returns the same error. More information here — https://www.postgresql.org/docs/9.5/static/functions-json.html.

> Note: In json_populate_record, json_populate_recordset, json_to_record and json_to_recordset, type coercion from the JSON is "best effort" and may not result in desired values for some types.

It does not seem to be a bug with `pg-types` though. After investigating this I looks like it only convert pg types to javascript types but maybe I am wrong.

How should we go about this now? If we go the route of converting it to postgres tupe directly how would you approach this?